File: app/src/main/java/com/improve10x/chatwithv2/templates/EditTemplateActivity.java
package com.improve10x.chatwithv2.templates;

import androidx.annotation.NonNull;

import android.os.Bundle;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.FirebaseFirestore;
import com.improve10x.chatwithv2.base.BaseActivity;
import com.improve10x.chatwithv2.databinding.ActivityEditTemplateBinding;

public class EditTemplateActivity extends BaseActivity {

    private ActivityEditTemplateBinding binding;
    private Template template;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityEditTemplateBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        if (getIntent().hasExtra("templates")) {
            getSupportActionBar().setTitle("Edit Template");
            template = (Template) getIntent().getSerializableExtra("templates");
            showData();
            handleEditBtn();
        }
    }

    private void showData() {
        binding.titleTxt.setText(template.titleText);
        binding.messageTxt.setText(template.messageText);
    }

    private void editTemplate(String id, String messageTxt, String titleTxt) {
        template = new Template(messageTxt, titleTxt);
        FirebaseFirestore db = FirebaseFirestore.getInstance();
        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
        db.collection("/users/" + user.getUid() + "/templates")
                .document(id)
                .set(template)
                .addOnSuccessListener(new OnSuccessListener<Void>() {
                    @Override
                    public void onSuccess(Void unused) {
                        showToast("Successfully edited the Template");
                        finish();
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        showToast("Failed to edit the Template");
                    }
                });
    }

    private void handleEditBtn() {
        binding.editBtn.setOnClickListener(view -> {
            String messageTxt = binding.messageTxt.getText().toString();
            String titleTxt = binding.titleTxt.getText().toString();
            editTemplate(template.id, messageTxt, titleTxt);
        });
    }
}

File: app/src/main/java/com/improve10x/chatwithv2/templates/EditTemplateActivity.kt
package com.improve10x.chatwithv2.templates

import android.os.Bundle
import android.os.PersistableBundle
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import com.improve10x.chatwithv2.base.BaseActivity
import com.improve10x.chatwithv2.base.showToast
import com.improve10x.chatwithv2.databinding.ActivityEditTemplateBinding

class EditTemplateActivity : AppCompatActivity() {

    private lateinit var binding: ActivityEditTemplateBinding
    private var template: Template? = null

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        binding = ActivityEditTemplateBinding.inflate(layoutInflater)
        setContentView(binding.root)
        if(intent.hasExtra("templates")) {
            supportActionBar?.title = "Edit Template"
            template = intent.getSerializableExtra("templates") as Template
            showData()
            handleEditBtn()
        }
    }

    private fun handleEditBtn() {
        binding.editBtn.setOnClickListener {
            val messageText = binding.messageTxt.text.toString()
            val titleText = binding.titleTxt.text.toString()
            template?.id?.let {
                editTemplate(it, messageText, titleText)
            }
        }
    }

    private fun editTemplate(id: String, messageText: String, titleText: String) {
        template = Template(messageText, titleText)
        val db = FirebaseFirestore.getInstance()
        val user = FirebaseAuth.getInstance().currentUser
        user?.let {
            db.collection("/users/" + it.uid + "/templates")
                .document(id)
                .set(template!!)
                .addOnSuccessListener {
                    showToast("Successfully edited the Template")
                    finish()
                }.addOnFailureListener {
                    showToast("Failed to edit the Template")
                }
        }
    }

    private fun showData() {
        template?.let {
            binding.titleTxt.setText(it.titleText)
            binding.messageTxt.setText(it.messageText)
        }
    }
}